### PR TITLE
[Android] Another bucket of fixes.

### PR DESCRIPTION
**Pull request by kumpera**

### CODE REVIEW

File: mcs/build/profiles/monodroid.make [hunk 1]
```diff
@@ -29,6 +29,9 @@ PROFILE_MCS_FLAGS = \
 	-nowarn:1699 \
 	-nostdlib \
 	$(DEFAULT_REFERENCES) \
+	-d:NO_SYSTEM_DRAWING_DEPENDENCY \
```
Comment (marek-safar) [inline mcs/build/profiles/monodroid.make]: This should not be needed, we do it in relevant Makefile

Comment (kumpera) [inline mcs/build/profiles/monodroid.make]: I copied this from XM to get system.runtime.remoting working.
I'll move that to the right makefile then.

File: mcs/class/Makefile [hunk 2]
```diff
@@ -105,6 +105,8 @@ monodroid_dirs := \
 	$(mobile_common_dirs)	\
 	Mono.CompilerServices.SymbolWriter	\
 	Mono.Btls.Interface	\
+	System.Runtime.Serialization.Formatters.Soap	\
+	System.Runtime.Remoting	\
```
Comment (marek-safar) [inline mcs/class/Makefile]: I am not sure this is good idea. IIRC XA ships whatever is in output folder so it'll hard to take this back

Comment (jonpryor) [inline mcs/class/Makefile]: > XA ships whatever is in output folder

No longer true in the xamarin-android repo. We [explicitly list the assemblies we copy](https://github.com/xamarin/xamarin-android/blob/master/build-tools/mono-runtimes/ProfileAssemblies.projitems).

This doesn't answer whether or not we *should* include these assemblies in the Xamarin.Android profile. @kumpera tells me that "without [them] remoting is completely broken." I'm not sure how important that *actually* is, especially considering that iOS doesn't ship these assemblies either.

All I'm reasonably certain of is that we *do* support AppDomains, insofar  as we're able to allocate them and invoke code within the AppDomains, and I "know" that Remoting is *somehow* involved in AppDomains...so how "broken" are we, really, and does it matter?

Comment (kumpera) [inline mcs/class/Makefile]: Without remoting, only the most primitive forms of cross-AppDomain communications work. It's required for transparent proxies and MarshalByRef objects.

We explicitly don't support AppDomains or Remoting on iOS.
Android was left as a maybe and we end up deciding to support AppDomains.

Comment (jonpryor) [inline mcs/class/Makefile]: Just to wrap up: this change is fine. It won't implicitly alter the assemblies included in Xamarin.Android.

Comment (kumpera) [inline mcs/class/Makefile]: We need SRR if we want AppDomains to work beyond trivial examples as most forms of cross-domain communications need it.

At this stage, it's ok if XA doesn't ship those two assemblies.

Additionally, given that this assembly is loaded by corlib using reflection, there's further work in tooling to make it possible to ship it without impacting our users.

File: mcs/class/Mono.Posix/Mono.Unix.Android/AndroidUtils.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
+using System;
+using System.Reflection;
+
+// using Android.Runtime;
+// using Android.Util;
+
+namespace Mono.Unix.Android
+{
+	internal sealed class AndroidUtils
```
Comment (marek-safar) [inline mcs/class/Mono.Posix/Mono.Unix.Android/AndroidUtils.cs]: Should be static

File: mcs/class/Makefile (hunk 2 above)
Comment (marek-safar) [inline mcs/class/Makefile]: I'd prefer not to add it, if nobody complained in 5 years or so why to do the extra work and lock ourself into supporting this forever

File: mcs/class/Mono.Posix/Mono.Unix.Android/AndroidUtils.cs (hunk 3 above)
Comment (kumpera) [inline mcs/class/Mono.Posix/Mono.Unix.Android/AndroidUtils.cs]: This is from XA, I didn't change it.

**Comment by kumpera** [comment]
Closing this PR and splitting it in runtime, BCL and test suites PRs so I can get faster turnaround.

**Pull request closed without merging**